ChurchCRM/CRM
# Bugfix: Add birthdate validation on data entry and CSV export

#### What's this PR do?
1. Add birthdate validation in **PersonEditor.php** preventing the following scenarios:
   1. Entering a birth year by itself (no day or month)
   2. Entering a month with no day (with or without a year)
   3. Entering a day with no month (with or without a year)
1. Modify **CSVCreateFile.ph**p such that a null year will export no year value rather than "0000-".

#### Screenshots (if appropriate)
**Changes to PersonEditor.php Behavior**
![image](https://user-images.githubusercontent.com/23409144/116010137-41840200-a5eb-11eb-9795-c3cca8e20cd2.png)
![image](https://user-images.githubusercontent.com/23409144/116010145-4f398780-a5eb-11eb-99c7-653e285067bc.png)
![image](https://user-images.githubusercontent.com/23409144/116010161-61b3c100-a5eb-11eb-8538-3ade5c20cb1d.png)
**Changes to CSVCreateFile.php Behavior**
![image](https://user-images.githubusercontent.com/23409144/116010317-3c738280-a5ec-11eb-9e33-c58031ea5524.png)

#### What Issues does it Close?

Closes #5719 

#### What are the relevant tickets?

#### Any background context you want to provide?
There had been very little data validation on birthdates when creating new people entries.  This fix prevents someone from only entering a year, or from entering a month without a day (or vice-versa).

Additionally, the CSV Export feature will not render a birth year if it is not provided, instead of rendering `0000-`

#### Where should the reviewer start?

#### How should this be manually tested?
**Testing the birthdate validation**
1. Log into your instance of ChurchCRM.
2. Choose _People ->Add New Person_.
3. Set the _Birth Month_ to _January_ and choose _Save_ - you should receive the `Invalid Birth Date: Missing birth month or day` error message.
4. Set the _Birth Month_ to _January_ and choose _Save_ - you should receive the `Invalid Birth Date: Missing birth month or day` error message.
5. Clear the _Birth Month_ value to its original _Select Month_ setting.
6. Set the _Birth Day_ to _1_ and choose _Save_ - you should receive the `Invalid Birth Date: Missing birth month or day` error message.
7. Clear the _Birth Day_ value to its original _Select Day_ setting.
8. Set the _Birth Year_ to _2021_ and choose _Save_ - you should receive the `Invalid Birth Date: Missing birth month and day` error message.

**Testing the birthdate validation**

1. Log into your instance of ChurchCRM.
2. Choose _People ->Add New Person_.
3. Enter a first and last name.
4. Provide a birth month and birth day, but leave the birth year empty.
5. Choose _Save_.
6. Choose _Add to Cart_.
7. In the top navigation bar, select the Shopping Cart icon (![image](https://user-images.githubusercontent.com/23409144/116010755-11d6f900-a5ef-11eb-8aec-1b65bbe942fd.png)) and choose _View Cart_.
8. Choose _CSV Export_.
9. Select the _Birth/Anniversary Date_ checkbox.
10. Choose _Create File_.  The result should display the birth day and month, but not the year.

Example CSV Output:
`"Family ID ","Person Id","Last Name","First Name","Address 1","Address 2","City","State","Zip","Country","Birth Date"`
`"","333","User","Test","","","","","","Canada","4-25"`

#### How should the automated tests treat this?

#### Questions:
- Is there a related website / article to substantiate / explain this change?

- Does the development wiki need an update?

- Does the user documentation wiki need an update?

- Does this add new dependencies?

- Does this need to add new data to the demo database
